### ISSUE TITLE
After Sign-In  Onboarding page on appearing . Directly homepage comes

### ISSUE THREAD

**Issue by jaydeeprana049** [adrianhajdin/threads#15]

**Comment by AymanBoumait** [comment]
go to your .env.local and update your after sign in to 
`NEXT_PUBLIC_CLERK_AFTER_SIGN_IN_URL=/onboarding`

**Comment by rayvinith** [comment]
after sign in nothing is given so how will onboarding appear

**Comment by kashanhz999** [comment]
Has anyone got the solution ?

**Comment by rajdevkumar2003** [comment]
same problem, did you find any solution yet?

**Comment by rajdevkumar2003** [comment]
> `export default authMiddleware({ publicRoutes:['/','/api/webhook/clerk','/onboarding'], ignoredRoutes:['/api/webhook/clerk'] });` add "/onboarding" to publicRoutes.

not working

**Comment by kashanhz999** [comment]
Share the screenshot of middleware.ts file

**Comment by rajdevkumar2003** [comment]
> Share the screenshot of middleware.ts file

here it is and the .env file is also the same as adrian shown us

![Screenshot (26)](https://github.com/adrianhajdin/threads/assets/136457828/1de2aa8a-6812-41aa-8f4d-6141e1b46707)

**Comment by rajdevkumar2003** [comment]
plz if anyone get any solution for it then reply

**Comment by rajdevkumar2003** [comment]
> Share the screenshot of middleware.ts file

is your deployed app is working fine as it should be ...?

**Comment by kashanhz999** [comment]
> > Share the screenshot of middleware.ts file
> 
> is your deployed app is working fine as it should be ...?

I haven't deployed it yet

**Comment by rajdevkumar2003** [comment]
> 

bro did you find any solution?

**Comment by kashanhz999** [comment]
No, for me this`export default authMiddleware({ publicRoutes:['/','/api/webhook/clerk','/onboarding'], ignoredRoutes:['/api/webhook/clerk'] });` worked. 

have you push your code to any repository??

**Comment by AymanBoumait** [comment]
I will give you the whole file as I have it

```
import { authMiddleware } from "@clerk/nextjs";

// This example protects all routes including api/trpc routes
// Please edit this to allow other routes to be public as needed.
// See https://clerk.com/docs/nextjs/middleware for more information about configuring your middleware
export default authMiddleware({
    publicRoutes: ['/', '/api/webhook/clerk'],
    ignoredRoutes: ['/api/webhook/clerk'],
});

export const config = {
  matcher: ["/((?!.*\\..*|_next).*)", "/", "/(api|trpc)(.*)"],
};

```

**Comment by AymanBoumait** [comment]
and just to be sure, what is the exact issue you are facing?

**Comment by tpratt57** [comment]
I am also having something of a similar issue. Onboarding did not appear for me after the sign-up and I am not given any functionality to sign out even though it's in the code. 
![Desktop Screenshot 2023 08 23 - 14 36 08 07](https://github.com/adrianhajdin/threads/assets/104174101/90eb8f2f-9f64-41e2-b222-e4ab1e4087ca)
![Desktop Screenshot 2023 08 23 - 14 36 48 90](https://github.com/adrianhajdin/threads/assets/104174101/61bc30c8-4406-4fb2-9e3e-a2cf37ff138f)
![Desktop Screenshot 2023 08 23 - 14 36 57 73](https://github.com/adrianhajdin/threads/assets/104174101/928c6c9a-b726-4ded-98d4-656a6b9528ce)
![Desktop Screenshot 2023 08 23 - 14 37 13 02](https://github.com/adrianhajdin/threads/assets/104174101/7a4ba096-e4a1-4d3b-98e6-5c35a2097231)

**Comment by tpratt57** [comment]
never mind, it seems to only be appearing for phone dimensions

**Comment by Ankan-cyber** [comment]
why you need onboarding after sign in, onboarding should appear after signup that's the fundamental relevance of onboarding

**Comment by Himanshu-370** [comment]
when I sign-up,  I don't get redirected to /onboarding page and also when I go on to /create-thread, /onboarding page appears and on filling details I am directed to home page. This happens everytime I go to /create-thread. I cannot check out the /create-thread page

**Comment by Ankan-cyber** [comment]
Just add the /onboarding in after signup in environment variable
